### PR TITLE
chore: bump cdk version

### PR DESCRIPTION
**Pull request by sobolk**
#### Description of changes

This PR bumps CDK version from 2.44.0 to 2.53.0.

I also tried 2.54.0 but that was quite a big disappointment. It messed up mock e2e and added quite unexpected cfn template parameters at runtime in `model-relational-with-key-transformer.e2e.test.ts` when debugging.

In order to make this happen `stack-synthetizer` requires changes to account for breaking change in protected APIs introduced by https://github.com/aws/aws-cdk/commit/8b2b38187b709a4e9a37a4de043a84267a9ec937#diff-071bf4af4f62cf68b345e393cdb18d584e85f49eea97b34a5319f57dfa64e22a

More details in former PR that locked CDK at 2.44.0 here https://github.com/aws-amplify/amplify-category-api/pull/923 .

#### Description of how you validated changes

E2E tests are passing except migration tests.

Migration tests are failing due to `testingWithLatestCodebase = true` and CDK mismatch and breaking change. Other tests are fine as they end up using older CDK from CLI at runtime (i.e. they call into API that exists both on api and cli synthesizers).
What happens is that types from `amplify-provider-awscloudformation` are still on older CDK and lack new protected APIs. Which in runtime leads to unexpected results. I.e. `RootStackSynthesizer` from cli side lacks the new API.
Please see content of this PR to understand break at StackSynthesizer protected APIs surface.

I made some of migration tests passing locally by directy editing `node_modules/amplify-provider-awscloudformation/lib/root-stack-builder/stack-synthesizer.js` and adding missing API to `RootStackSynthesizer`.
That gives pretty high confidence that when we align CDK on both sides this problem will disappear.

**In order to clear those test failures we need to bump CDK in both repositories and complete release loop.**

Example of local edits. Auth test is failing because I have not edited edited `AuthStackSynthesizer`.

image

After fixing `AuthStackSynthesizer` manually:

image

Error caused by mismatch looks like this (i.e. without fixing the problem locally)
```
Cannot read property 'Outputs' of undefined
TypeError: Cannot read property 'Outputs' of undefined
    at AmplifyRootStackTransform.synthesizeTemplates (/Users/sobkamil/git/amplify-category-api/node_modules/amplify-provider-awscloudformation/lib/root-stack-builder/root-stack-transform.js:127:40)
    at AmplifyRootStackTransform.transform (/Users/sobkamil/git/amplify-category-api/node_modules/amplify-provider-awscloudformation/lib/root-stack-builder/root-stack-transform.js:149:37)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
    at async formNestedStack (/Users/sobkamil/git/amplify-category-api/node_modules/amplify-provider-awscloudformation/lib/push-resources.js:642:17)
    at async generateAndUploadRootStack (/
Users/sobkamil/git/amplify-category-api/node_modules/amplify-provider-awscloudformation/lib/push-resources.js:885:25)
    at async Object.run (/Users/sobkamil/git/amplify-category-api/node_modules/amplify-provider-awscloudformation/lib/push-resources.js:190:17)
    at async Promise.all (index 0)
    at async providersPush (/Users/sobkamil/git/amplify-category-api/node_modules/amplify-cli-internal/lib/extensions/amplify-helpers/push-resources.js:109:5)
    at async AmplifyToolkit.pushResources [as _pushResources] (/Users/sobkamil/git/amplify-category-api/node_modules/amplify-cli-internal/lib/extensions/amplify-helpers/push-resources.js:83:13)
    at async Object.executeAmplifyCommand (/Users/sobkamil/git/amplify-category-api/node_modules/amplify-cli-internal/lib/index.js:172:9)
    at async executePluginModuleCommand (/Users/sobkamil/git/amplify-category-api/node_modules/amplify-cli-internal/lib/execution-manager.js:142:5)
    at async executeCommand (/Users/sobkamil/git/amplify-category-api/node_module
s/amplify-cli-internal/lib/execution-manager.js:40:9)
    at async Object.run (/Users/sobkamil/git/amplify-category-api/node_modules/amplify-cli-internal/lib/index.js:112:5)
```

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-graphql-transformer-core/src/cdk-compat/stack-synthesizer.ts [hunk 1]
```diff
@@ -12,18 +13,12 @@ export class TransformerStackSythesizer extends LegacyStackSynthesizer {
   private readonly mapingTemplateAssets: Map<string, string> = new Map();
   private _deploymentBucket?: CfnParameter;
   private _deploymentRootKey?: CfnParameter;
-  private boundStack?: Stack;
```
Comment (sobolk) [inline packages/amplify-graphql-transformer-core/src/cdk-compat/stack-synthesizer.ts:15]: StackSynthesizer introduced `boundStack` property and similar handling as this type.

File: packages/amplify-graphql-transformer-core/src/cdk-compat/stack-synthesizer.ts [hunk 2]
```diff
@@ -37,6 +32,22 @@ export class TransformerStackSythesizer extends LegacyStackSynthesizer {
     );
   }
 
+  protected synthesizeTemplate(session: ISynthesisSession, _?: string): FileAssetSource {
+    const stack = this.boundStack;
```
Comment (sobolk) [inline packages/amplify-graphql-transformer-core/src/cdk-compat/stack-synthesizer.ts:36]: this method replaced `synthesizeStackTemplate` in runtime.

File: packages/amplify-graphql-transformer-core/src/cdk-compat/stack-synthesizer.ts [hunk 3]
```diff
@@ -69,8 +80,6 @@ export class TransformerStackSythesizer extends LegacyStackSynthesizer {
    * addFileAsset
    */
   public addFileAsset(asset: FileAssetSource): FileAssetLocation {
-    assertNotNull(this.boundStack);
-
```
Comment (sobolk) [inline packages/amplify-graphql-transformer-core/src/cdk-compat/stack-synthesizer.ts:73]: this is no longer needed StackSynthesizer.boundStack throws if property is not set.

File: packages/amplify-graphql-transformer-core/src/cdk-compat/stack-synthesizer.ts [hunk 4]
```diff
@@ -12,18 +13,12 @@ export class TransformerStackSythesizer extends LegacyStackSynthesizer {
...
-  public bind(stack: Stack): void {
-    this.boundStack = stack;
-    super.bind(stack);
-  }
 
   /**
    * synthesizeStackTemplate
+   *
+   * This method has been deprecated by cdk and is not used in runtime.
+   * It got replaced by synthesizeTemplate.
```
Comment (akshbhu) [inline packages/amplify-graphql-transformer-core/src/cdk-compat/stack-synthesizer.ts:21]: /**
 * @deprecated Use synthesizeTemplate instead
 */

File: packages/amplify-category-api/src/__tests__/provider-utils/awscloudformation/ecs-apigw-stack.test.ts [hunk 5]
```diff
@@ -19,7 +19,7 @@ describe('ecs stack', () => {
           undefined,
           undefined,
           undefined,
-          undefined,
+          'testImage',
```
Comment (akshbhu) [inline packages/amplify-category-api/src/__tests__/provider-utils/awscloudformation/ecs-apigw-stack.test.ts:22]: why is this needed ?

**Review by akshbhu — APPROVED** [review]

File: packages/amplify-category-api/src/__tests__/provider-utils/awscloudformation/ecs-apigw-stack.test.ts (hunk 5 above)
Comment (sobolk) [reply packages/amplify-category-api/src/__tests__/provider-utils/awscloudformation/ecs-apigw-stack.test.ts:22]: CDK added validation for that property to be present. Sigh... another technically breaking change.

File: packages/amplify-graphql-transformer-core/src/cdk-compat/stack-synthesizer.ts (hunk 4 above)
Comment (sobolk) [reply packages/amplify-graphql-transformer-core/src/cdk-compat/stack-synthesizer.ts]: I added deprecation marker, but I keep wording. Technically we're not "using it" but CDK calls into it at runtime.

File: packages/amplify-graphql-transformer-core/src/cdk-compat/stack-synthesizer.ts (hunk 2 above)
Comment (phani-srikar) [reply packages/amplify-graphql-transformer-core/src/cdk-compat/stack-synthesizer.ts:36]: What does replace mean? Is this a copy/clone of some CDK method? 
Also, I don't see any usages of this new method. Is this a place holder?

**Review by phani-srikar — CHANGES_REQUESTED** [review]

**Review by alharris-at — APPROVED** [review]

**Review by phani-srikar — APPROVED** [review]
